### ISSUE TITLE
Printing issue with ModelingToolkit

### ISSUE THREAD

**Issue by dpsanders** [JuliaDiff/TaylorSeries.jl#233]
It would nice to be able to do symbolic calculations using ModelingToolkit.jl. 
Construction seems to work but there's an error in printing:

```jl
julia> using TaylorSeries, ModelingToolkit

julia> @variables x[1:5]
(Operation[x₁, x₂, x₃, x₄, x₅],)

julia> y = Taylor1(x);

julia> y
Error showing value of type Taylor1{Operation}:
ERROR: MethodError: no method matching pretty_print(::Taylor1{Operation})
```

**Comment by dpsanders** [comment]
e.g.
```
julia> z = y * y;

julia> z.coeffs
5-element Array{Operation,1}:
                                               x₁ * x₁
                                     x₁ * x₂ + x₂ * x₁
                         (x₁ * x₃ + x₂ * x₂) + x₃ * x₁
             ((x₁ * x₄ + x₂ * x₃) + x₃ * x₂) + x₄ * x₁
 (((x₁ * x₅ + x₂ * x₄) + x₃ * x₃) + x₄ * x₂) + x₅ * x₁
```
(Note that ModelingToolkit currently has no simplification.
Unfortunately I don't think we can really use TaylorN for this if there are too many new variables.)

**Comment by lbenet** [comment]
Interesting idea... I notice that whatever seems to work is because `Operation <: Number`.

What I don't quite see is how to exploit the `x` variable(s) when they are coefficients of a Taylor polynomial. Or, are you thinking in propagating the actual coefficients, to somehow exploit them independently?

**Comment by dpsanders** [comment]
I'm just thinking about obtaining symbolic expressions for the coefficients of the result of a calculation with Taylor series.

**Comment by lbenet** [comment]
```julia
julia> using TaylorSeries, ModelingToolkit

julia> use_show_default(true) # pretty_print doesn't work for Taylor1{Operations}
true

julia> @parameters x[0:3]   # I don't know if @variables is better here
(Operation[x₀(), x₁(), x₂(), x₃()],)

julia> xt = Taylor1(x)
Taylor1{Operation}(Operation[x₀(), x₁(), x₂(), x₃()], 3)

julia> xt * xt
Taylor1{Operation}(Operation[x₀() * x₀(), x₀() * x₁() + x₁() * x₀(), (x₀() * x₂() + x₁() * x₁()) + x₂() * x₀(), ((x₀() * x₃() + x₁() * x₂()) + x₂() * x₁()) + x₃() * x₀()], 3)

julia> getcoeff(ans, 2)
(x₀() * x₂() + x₁() * x₁()) + x₂() * x₀()
```

Is this more or less what you have in mind?

**Comment by dpsanders** [comment]
Yes exactly. Yes I think `@variables` is better since you don't get the parentheses. (Or maybe that's just to do with the version of ModelingToolkit.)

**Comment by lbenet** [comment]
With `@variables` or `@parameters` I still get the parenthesis; maybe it is a question of the version. I am using ModelingToolkit v0.8.0.

Regarding other functions, things become brittle very fast, but could be fixed. The problem I just noticed is that `NumbersNotSeries` is not recognizing the types introduced by ModelingToolkit.

**Comment by ChrisRackauckas** [comment]
You can get rid of the parentheses by calling simplify_expr

**Comment by lbenet** [comment]
> You can get rid of the parentheses by calling simplify_expr

Very nice, thanks!

**Comment by lbenet** [comment]
> I'm just thinking about obtaining symbolic expressions for the coefficients of the result of a calculation with Taylor series.

I just realized that the following is another alternative, using TaylorSeries only:
```julia
julia> using TaylorSeries

julia> displayBigO(false)
false

julia> x = set_variables("x₀ x₁ x₂ x₃ x₄", order=10);

julia> x0, x1, x2, x3, x4 = x;

julia> xt = Taylor1(x, 4)
  1.0 x₀ + ( 1.0 x₁) t + ( 1.0 x₂) t² + ( 1.0 x₃) t³ + ( 1.0 x₄) t⁴ 

julia> xt * xt
  1.0 x₀² + ( 2.0 x₀ x₁) t + ( 2.0 x₀ x₂ + 1.0 x₁²) t² + ( 2.0 x₀ x₃ + 2.0 x₁ x₂) t³ + ( 2.0 x₀ x₄ + 2.0 x₁ x₃ + 1.0 x₂²) t⁴ 

julia> sin(xt)
  1.0 x₀ - 0.16666666666666666 x₀³ + 0.008333333333333333 x₀⁵ - 0.00019841269841269839 x₀⁷ + 2.7557319223985884e-6 x₀⁹ + ( 1.0 x₁ - 0.5 x₀² x₁ + 0.041666666666666664 x₀⁴ x₁ - 0.0013888888888888887 x₀⁶ x₁ + 2.4801587301587298e-5 x₀⁸ x₁) t + ( 1.0 x₂ - 0.5 x₀² x₂ - 0.5 x₀ x₁² + 0.041666666666666664 x₀⁴ x₂ + 0.08333333333333333 x₀³ x₁² - 0.0013888888888888887 x₀⁶ x₂ - 0.004166666666666667 x₀⁵ x₁² + 2.4801587301587298e-5 x₀⁸ x₂ + 9.920634920634919e-5 x₀⁷ x₁²) t² + ( 1.0 x₃ - 0.5 x₀² x₃ - 1.0 x₀ x₁ x₂ - 0.16666666666666666 x₁³ + 0.041666666666666664 x₀⁴ x₃ + 0.16666666666666666 x₀³ x₁ x₂ + 0.08333333333333333 x₀² x₁³ - 0.0013888888888888885 x₀⁶ x₃ - 0.008333333333333333 x₀⁵ x₁ x₂ - 0.006944444444444444 x₀⁴ x₁³ + 2.4801587301587298e-5 x₀⁸ x₃ + 0.00019841269841269839 x₀⁷ x₁ x₂ + 0.00023148148148148144 x₀⁶ x₁³) t³ + ( 1.0 x₄ - 0.5 x₀² x₄ - 1.0 x₀ x₁ x₃ - 0.5 x₀ x₂² - 0.5 x₁² x₂ + 0.041666666666666664 x₀⁴ x₄ + 0.16666666666666666 x₀³ x₁ x₃ + 0.08333333333333333 x₀³ x₂² + 0.25 x₀² x₁² x₂ + 0.041666666666666664 x₀ x₁⁴ - 0.0013888888888888887 x₀⁶ x₄ - 0.008333333333333333 x₀⁵ x₁ x₃ - 0.004166666666666667 x₀⁵ x₂² - 0.020833333333333332 x₀⁴ x₁² x₂ - 0.006944444444444444 x₀³ x₁⁴ + 2.4801587301587298e-5 x₀⁸ x₄ + 0.00019841269841269839 x₀⁷ x₁ x₃ + 9.920634920634919e-5 x₀⁷ x₂² + 0.0006944444444444444 x₀⁶ x₁² x₂ + 0.0003472222222222222 x₀⁵ x₁⁴) t⁴ 
```

I guess it has its own limitations, though you can get some results for more interesting functions (e.g., `sin(xt)`).

**Comment by dpsanders** [comment]
Yes I did think of that, but that won't work for many variables I guess.

**Comment by lbenet** [comment]
I guess you are right, though maybe a combination of nesting Taylor1's and TaylorN may work... I truly never though about such a beast.

**Comment by lbenet** [comment]
Yet, it does work with `HomogeneousPolynomial`s:
```julia
julia> x[1]([xx, yy])
0 + 1.0 * (xx ^ 1 * yy ^ 0)
```

So the problem may be related to the Horner implementation.

**Comment by lbenet** [comment]
This is solved by #244:
```julia
julia> using TaylorSeries, ModelingToolkit

julia> @variables x[0:4]
(Operation[x₀, x₁, x₂, x₃, x₄],)

julia> y = Taylor1(x);

julia> y
 x₀ + x₁ t + x₂ t² + x₃ t³ + x₄ t⁴ + 𝒪(t⁵)

julia> 2*y
 2x₀ + 2x₁ t + 2x₂ t² + 2x₃ t³ + 2x₄ t⁴ + 𝒪(t⁵)

julia> y*y
 x₀ * x₀ + x₀ * x₁ + x₁ * x₀ t + (x₀ * x₂ + x₁ * x₁) + x₂ * x₀ t² + ((x₀ * x₃ + x₁ * x₂) + x₂ * x₁) + x₃ * x₀ t³ + (((x₀ * x₄ + x₁ * x₃) + x₂ * x₂) + x₃ * x₁) + x₄ * x₀ t⁴ + 𝒪(t⁵)
```

And  you can play further, e.g., `sqrt(1+y)`.
